src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-details/rgw-bucket-details.component.ts

```diff
@@ -11,15 +13,53 @@ export class RgwBucketDetailsComponent implements OnChanges {
   @Input()
   selection: any;
 
+  aclPermissions: Record<string, string[]> = {};
+
   constructor(private rgwBucketService: RgwBucketService) {}
 
   ngOnChanges() {
     if (this.selection) {
       this.rgwBucketService.get(this.selection.bid).subscribe((bucket: object) => {
         bucket['lock_retention_period_days'] = this.rgwBucketService.getLockDays(bucket);
         this.selection = bucket;
-        this.selection.bucket_policy = JSON.parse(this.selection.bucket_policy) || {};
+        this.aclPermissions = this.parseXmlAcl(this.selection.acl, this.selection.owner);
       });
     }
   }
+
+  parseXmlAcl(xml: any, bucketOwner: string): Record<string, string[]> {
+    const parser = new xml2js.Parser({ explicitArray: false, trim: true });
+    let data: Record<string, string[]> = {
+      Owner: ['-'],
+      AllUsers: ['-'],
+      AuthenticatedUsers: ['-']
+    };
+    parser.parseString(xml, (err, result) => {
+      if (err) {
+        return null;
+      }
+      const xmlGrantees: any = result['AccessControlPolicy']['AccessControlList']['Grant'];
+      if (Array.isArray(xmlGrantees)) {
+        for (let i = 0; i < xmlGrantees.length; i++) {
+          const grantee = xmlGrantees[i];
+          if (grantee.Grantee.URI) {
+            const granteeGroup = grantee.Grantee.URI.split('/').pop();
+            if (data[granteeGroup].includes('-')) {
+              data[granteeGroup] = [grantee.Permission];
+            } else {
+              data[granteeGroup].push(grantee.Permission);
```

Comment:
Just a suggestion, you many leave it...
We can perhaps move this into  a smaller function to improve readability and debugging of code..
Its a lot going on here...

Reply:
Since it is just one thing this function is doing, parsing the xml string to the formated data, I thought it was okay to keep like that, I agree is kinda too much, I tried to keep it more readable with the types but not sure if that even helped.

How would you split this function? @afreen23

Reply:
Np! may leave it if it does not makes sense... I dont have much idea with the acl logic hence cant help much..
Initially I was of the thinking to do perhaps like: 
```
addGranteeGroup(data) {
 const granteeGroup = grantee.Grantee.URI.split('/').pop();
            if (data[granteeGroup].includes('-')) {
              data[granteeGroup] = [grantee.Permission];
            } else {
              data[granteeGroup].push(grantee.Permission);
        return data;
}
```

Thanks but I want you take it as suggestion!

Reply:
Thanks for the explanation, that really helpful!